test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -162,19 +162,11 @@ call_log_manager::process_call_log_entry(json call_log_entry)
             /* Get the session from the session map. */
             session_simulator *session = _session_map.at(session_id);
 
-            /*
-             * Check no other transactions are running. There should be a 1:1 relationship between
-             * session and transaction.
-             */
-            if (session->is_txn_running()) {
+            if (session->begin_transaction() != 0) {
                 std::cerr << "Could not begin transaction, session already has a running "
```

Comment:
I would move this err log in the `session_simulator`. The caller may not know why we could not begin the transaction. In the future, we could prevent from starting a new transaction for other reasons if that makes sense.

Reply:
That's true.